grayskull/pypi/pypi.py

```diff
@@ -928,11 +965,12 @@ def _get_run_req_from_requires_dist(self, requires_dist: List) -> List:
                     selector = f"  # [{selector}]"
                 else:
                     selector = ""
-            pkg_name, version = PyPi._get_name_version_from_requires_dist(
-                list_raw_requirements[0]
-            )
-            run_req.append(f"{pkg_name} {version}{selector}".strip())
-        return run_req
+
+                pkg_name, version = PyPi._get_name_version_from_requires_dist(
+                    list_raw_requirements[0]
+                )
+                run_req.append(f"{pkg_name} {version}{selector}".strip())
+                return run_req
```

Comment:
This should go into the upper indentation level; otherwise we are returning early after only the first iteration.